src/Logger_int.java

```diff
@@ -0,0 +1,340 @@
+import java.util.concurrent.locks.LockSupport;
+
+/**
+ * @author Andrey Pavlenko
+ */
+public class Logger {
+    private final PrintWriter _logWriter;
+    private final List<PBar> _pbar = new ArrayList<PBar>(3);
+    private volatile Level _level;
+    private volatile boolean _progressBarSupported;
+
+    public Logger(final OutputStream logWriter,
+            final boolean progressBarSupported) {
+        this(logWriter, progressBarSupported, null);
+    }
+
+    public Logger(final OutputStream logWriter,
+            final boolean progressBarSupported, final Level level) {
+        this(new PrintWriter(logWriter), progressBarSupported, level);
+    }
+
+    public Logger(final PrintWriter logWriter,
+            final boolean progressBarSupported) {
+        this(logWriter, progressBarSupported, null);
+    }
+
+    public Logger(final PrintWriter logWriter,
+            final boolean progressBarSupported, final Level level) {
+        _logWriter = logWriter;
+        _progressBarSupported = progressBarSupported;
+
+        if (level == null) {
+            if ("true".equalsIgnoreCase(System.getenv("GITST_DEBUG"))) {
+                _level = Level.DEBUG;
+            } else {
+                _level = Level.INFO;
+            }
+        } else {
+            _level = level;
+        }
+    }
+
+    public static Logger createConsoleLogger(final Level level) {
+        final Console c = System.console();
+        final String env = System.getenv("GITST_PB");
+        final boolean enable = "true".equalsIgnoreCase(env);
+        final boolean disable = "false".equalsIgnoreCase(env);
+
+        if (c != null) {
+            return new Logger(c.writer(), !disable, level);
+        } else {
+            return new Logger(System.out, enable, level);
+        }
+    }
+
+    public boolean isProgressBarSupported() {
+        return _progressBarSupported;
+    }
+
+    public void setProgressBarSupported(final boolean progressBarSupported) {
+        _progressBarSupported = progressBarSupported;
+    }
+
+    public Level getLevel() {
+        return _level;
+    }
+
+    public void setLevel(final Level level) {
+        _level = level;
+    }
+
+    public void debug(final Object msg) {
+        if (isDebugEnabled()) {
+            log(msg);
+        }
+    }
+
+    public void info(final Object msg) {
+        if (isInfoEnabled()) {
+            log(msg);
+        }
+    }
+
+    public void warn(final Object msg) {
+        if (isWarnEnabled()) {
+            log(msg);
+        }
+    }
+
+    public void error(final Object msg) {
+        error(msg, null);
+    }
+
+    public void error(final Object msg, final Throwable ex) {
+        if (isErrorEnabled()) {
+            synchronized (this) {
+                if (!_pbar.isEmpty()) {
+                    clearPogress();
+                    _logWriter.println(msg);
+
+                    if (ex != null) {
+                        ex.printStackTrace(_logWriter);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=squid%3AS1148)